### PR TITLE
crimson/osd: Support Pool EIO flag

### PR DESCRIPTION
**Pull request by Matan-B**
```
$ CRIMSON_COMPAT=true RBD_FEATURES= bin/ceph_test_rados_api_aio_pp 
  --gtest_filter="LibRadosAio.*PoolEIOFlag" 
Running main() from gmock_main.cc
Note: Google Test filter = LibRadosAio.*PoolEIOFlag
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from LibRadosAio 
[ RUN      ] LibRadosAio.SimplePoolEIOFlag
[       OK ] LibRadosAio.SimplePoolEIOFlag (15634 ms) 
[ RUN      ] LibRadosAio.PoolEIOFlag  
setting pool EIO
waiting for 99,100
max_success 98, min_failed 99
[       OK ] LibRadosAio.PoolEIOFlag (16008 ms) 
[----------] 2 tests from LibRadosAio (31642 ms total)
[----------] Global test environment tear-down 
[==========] 2 tests from 1 test suite ran. (31642 ms total)  
[  PASSED  ] 2 tests.       
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [x] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/test/librados/aio_cxx.cc [hunk 1]
```diff
@@ -2283,6 +2287,30 @@ void pool_io_callback(completion_t cb, void *arg /* Actually AioCompletion* */)
...
+  AioTestDataPP test_data;
+  ASSERT_EQ("", test_data.init());
+
+    auto my_completion = std::unique_ptr<AioCompletion>{Rados::aio_create_completion()};
+    ASSERT_TRUE(my_completion);
+
+    bufferlist empty;
+	  ASSERT_EQ(0, test_data.m_cluster.mon_command(
+	    "{\"prefix\": \"osd pool set\", \"pool\": \"" + test_data.m_pool_name +
+	    "\", \"var\": \"eio\", \"val\": \"true\"}", empty, nullptr, nullptr));
```
Comment (tchaikov) [inline src/test/librados/aio_cxx.cc:2300]:
```suggestion
	    fmt::format(R"({
"prefix": "osd pool set",
"pool": "{}",
"var": "eio",
"val": "true"
})",
                        test_data.m_pool_name),
            empty, nullptr, nullptr));
```

**Comment by tchaikov** [comment]
jenkins test make check

**Review by liu-chunmei — APPROVED** [review]
LGTM.

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
sorry. i should have run the change against qa tests before merging it.

**Pull request merged**